### PR TITLE
[Docs Improvement] Tidy Database-* Entries

### PR DESCRIPTION
**Pull request by gdjohnson**
This branch tidies up database-related entries written by Nakul in 2023. It removes redundant or outdated information, adds pointers between related pages, and cleans up several formatting issues.

## Link to Issue
Closes: #8320

### CODE REVIEW

File: knowledge_base/Database-Monitoring.md [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
+## Heroku Database Monitoring
 
 To connect to the PSQL schema, use the following command:
 
 ```bash
 heroku pg:psql postgresql-clear-46785 -a commonwealth-staging
 ```
 
-## About the Database
+As of 230413, the current database being used is Postgres 13.10. You can retrieve general information and uptime using the following SQL queries:
```
Comment (timolegros) [inline knowledge_base/Database-Monitoring.md:22]: We are on version 15.7 now. Major version (15) will not change without a manual update but the minor version may be updated by Heroku.

File: knowledge_base/Database-Monitoring.md [hunk 2]
```diff
@@ -35,43 +44,24 @@ To inspect the definition of a particular object in the schema, use the followin
...
-pnpm --packages/commonwealth datadog-db-setup commonwealth-staging packages/commonwealth/datadog
+WHERE schemaname = 'public'
+ORDER BY tablename, indexname;
 ```
 
-## Datadog Postgres Dashboards
-
-There are two Dashboards available in Datadog for monitoring Postgres:
-
-### Postgres Metrics Dashboard
```
Comment (timolegros) [inline knowledge_base/Database-Monitoring.md:67]: We have Database monitoring enabled now so DB metrics/query information can be found here: https://us5.datadoghq.com/databases/list?dbType=PostgreSQL

File: knowledge_base/Database-Optimization.md [hunk 3]
```diff
@@ -1,24 +1,37 @@
 # Query Optimization
```
Comment (timolegros) [inline knowledge_base/Database-Optimization.md:1]: Honestly, I would delete this entire file. It covers very rudimentary optimizations that anyone working with the DB should know by heart. We are building much better processes now that include looking at query plans and load testing. This information provides little value and I highly doubt anyone has or will reference it. Optimizations happen on a case-by-case basis by thoroughly reviewing the query plan.

**Review by timolegros — CHANGES_REQUESTED** [review]

File: knowledge_base/Database-Monitoring.md (hunk 2 above)
Comment (gdjohnson) [reply knowledge_base/Database-Monitoring.md:67]: Moved all this out since redundancies with Datadog.md

**Comment by gdjohnson** [comment]
Thanks so much @timolegros 

Updated Postgres version info, and deleted the unnecessary `Database-Optimization.md` file

**Review by jnaviask — APPROVED** [review]

**Pull request merged**